<launch>

  <arg name="motor_type" default="0" />
  <arg name="angle_offset" default="0" />
  
  <group unless="$(arg motor_type)">
    <include file="$(find vesc_driver)/launch/vesc_drive_xycar_motor.launch" />
  </group>
  
  <!-- xycar_motor node 실행 -->
  <node pkg="xycar_motor" type="xycar_motor.py" name="xycar_motor" output="screen" >
    <param name="motor_type" value="$(arg motor_type)" />
    <param name="angle_offset" value="$(arg angle_offset)" />
  </node>

  <!-- usb_cam node 실행 -->
 
  <node pkg="usb_cam" type="usb_cam_node" name="usb_cam" output="screen" >
    <param name="video_device" value="/dev/videoCAM" />
    <param name="autoexposure" value="false" />
    <param name="exposure" value="160" />
    <param name="image_width" value="1280" />
    <param name="image_height" value="720" />
    <param name="pixel_format" value="yuyv" />
    <param name="io_method" value="mmap" />
    <param name="camera_name" value="usb_cam" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="camera_info_url" value="file://$(find usb_cam)/calibration/usb_cam.yaml" />
  </node>

	
 <!-- IMU Launching -->
 <!--
  <node pkg="xycar_imu" type="9dof_imu_node.py" name="xycar_imu" output="screen">
    <param name="port" type="string" value="/dev/ttyIMU" />
    <param name="frame_id" type="string" value="imu" />
    <param name="topic" type="string" value="imu" />
    <param name="queue_size" type="int" value="1" />
    <param name="hz" type="int" value="30" />
  </node>
  -->
  <!-- Lidar Launching -->
  <!-- YD Lidar -->
  <!--
  <node pkg="xycar_lidar" type="xycar_ydlidar" name="xycar_lidar" output="screen" respawn="false" >
    <param name="resolution_fixed" type="bool"  value="true"/>
    <param name="auto_reconnect"   type="bool"  value="true"/>
    <param name="port"         type="string" value="/dev/ttyLIDAR"/>  
    <param name="baudrate"     type="int"    value="230400"/>  
    <param name="frame_id"     type="string" value="laser_frame"/>
    <param name="reversion"    type="bool"   value="false"/>
    <param name="angle_min"    type="double" value="-180" />
    <param name="angle_max"    type="double" value="180" />
    <param name="range_min"    type="double" value="0.1" />
    <param name="range_max"    type="double" value="16.0" />
    <param name="ignore_array" type="string" value="" />
    <param name="frequency"    type="double" value="10"/>
    <param name="samp_rate"    type="int"    value="5"/>
  </node>
  -->
  <!-- RP Lidar not use! -->
  <!--
  <node pkg="xycar_lidar" type="xycar_rplidar" name="xycar_lidar" output="screen">
    <param name="serial_port"         type="string" value="/dev/ttyLIDAR"/>
    <param name="serial_baudrate"     type="int"    value="115200"/>
    <param name="frame_id"            type="string" value="laser_frame"/>
    <param name="inverted"            type="bool"   value="false"/>
    <param name="angle_compensate"    type="bool"   value="true"/>
  </node>
  -->
  <!-- Ultra Launching -->
  <!--
  <node pkg="xycar_ultrasonic" type="xycar_ultrasonic.py" name="xycar_ultrasonic" output="screen" />	
  -->
  
  <!-- xycar_joy node 실행 -->
  <node name="xycar_self2" pkg="xycar_self2" type="drive.py" output="screen"/>
  
</launch>

